app/assets/javascripts/discourse/app/models/group.js

```diff
@@ -375,7 +375,7 @@ const Group = RestModel.extend({
   },
 
   requestMembership(reason) {
-    return ajax(`/groups/${this.name}/request_membership`, {
+    return ajax(`/groups/${this.name}/request_membership.json`, {
```

Comment:
This simple change allows to display the proper error message from the server. (it doesn’t change the fact the UI currently won’t let you write more than 5k chars though)